server.js

```diff
@@ -1,32 +1,150 @@
+  }
+  res.send(myEndPoints(app))
 })
 
+app.get('/books', (req, res) => {
+  const books = booksData
+  const page = req.query.page ? req.query.page -1 : 0 //page 0 as default, page number is -1 compared to the array index
+    const pageSize = req.query.pageSize ?? 20 //default is 20
+    const startIndex = page * pageSize
+    const endIndex = startIndex + pageSize
+    const bookPerPage = books.slice(startIndex, endIndex)
+    const returnObj = {
+      amountOfData: books.length,
+      amounfOfPages: Math.ceil(books.length / pageSize),
+      currentPage: page + 1,
+      pageSize: pageSize,
+      startIndex: startIndex,
+      endIndex: endIndex,
+      results: bookPerPage
+    }
+
+    if (bookPerPage.length === 0) {
+      res.status(404).json(ERROR_DATA_NOT_FOUND)
+    } else {
+      res.json(returnObj)
+    }
+})
+
+//search for books by Id
+app.get('/books/:id', (req, res) => {
+  const bookId = req.params.id
+  const filterById = booksData.filter((book) => book.bookID === +bookId)
+  if (filterById.length === 0) {
+    res.status(404).json(ERROR_DATA_NOT_FOUND)
+  } else {
+    res.json(filterById) 
+  }
+})
+
+//search for books by author
+app.get('/books/authors/:author', (req, res) => {
+  const { author } = req.params
+  const filterByAuthor = booksData.filter((book) => {
+    const lowercaseAuthors = book.authors.toLowerCase()
+    return lowercaseAuthors.includes(author);
+  })
+
+  if (filterByAuthor.length === 0) {
+    res.status(404).json(ERROR_DATA_NOT_FOUND)
+  } else {
+    res.json(filterByAuthor)
+  }
+})
+
+//search for books by title
+app.get('/books/titles/:title', (req, res) => {
+  const { title } = req.params
+  const filterByTitle = booksData.filter((book) => {
+    const titleString = String(book.title)
+    const lowercaseTitle = titleString.toLowerCase()
+    return lowercaseTitle.includes(title);
+  })
+  if (filterByTitle.length === 0) {
+    res.status(404).json(ERROR_DATA_NOT_FOUND)
+  } else {
+    res.json(filterByTitle)
+  }
+})
+
+//search for books by language
+//take the first two character of the query and compare to the first two letter in the language_code
+app.get('/books/language/search', (req, res) => {
+  const language = req.query.lang
+  const splitLangInput = language.split('')
+  const sliceLangInput = splitLangInput.slice(0, 2)
+  
+  const filterByLanguage = booksData.filter((book) => {
+    const langString = String(book.language_code)
+    const lowercaseLang = langString.toLowerCase()
+    const splitLangData = lowercaseLang.split('')
+    const sliceLangData = splitLangData.slice(0, 2)
+
+    return sliceLangInput[0] === sliceLangData[0] && sliceLangInput[1]=== sliceLangData[1]
+  })
+
+  if (filterByLanguage.length === 0) {
+      return res.status(404).json(ERROR_DATA_NOT_FOUND)
+    } else {
+      res.json(filterByLanguage)
+    }
+})
+
+//sort books by average_rating
+app.get('/books/rating/sort', (req, res) => {
```

Comment:
maybe creating `/books/sort/:field` would make you create just one endpoint with the possibility to sort it as your user wants?